api/v1alpha1/api_key_auth_types.go

```diff
@@ -0,0 +1,59 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+import (
+	gwapiv1 "sigs.k8s.io/gateway-api/apis/v1"
+)
+
+const APIKeysSecretKey = "credentials"
+
+// APIKeyAuth defines the configuration for the API Key Authentication.
+type APIKeyAuth struct {
+	// Credentials is the Kubernetes secret which contains the API keys.
+	// This is an Opaque secret.
+	// Each API key is stored in the key representing the client id,
+	// which can be used in AllowedClients to authorize the client in a simple way.
+	Credentials gwapiv1.SecretObjectReference `json:"credentials"`
```

Comment:
prefer `CredentialRefs []gwapiv1.SecretObjectReference` we can start and only support a single entry in the list, but it helps us leave room for expansion

Reply:
> we can start and only support a single entry in the list

Any reason for this? It's not complicated to support the multiple secret refs from the first step.

Reply:
go for it, supporting multiple refs would be great ! didn't want to request another enhancement to this initial API